test/functional/find_tests.js

```diff
@@ -3121,4 +3122,46 @@ describe('Find', function() {
       });
     }
   });
+
+  it('should respect client-level read preference', {
+    metadata: { requires: { topology: ['replicaset'] } },
+
+    test: function(done) {
+      const config = this.configuration;
+      const client = config.newClient({}, { monitorCommands: true, readPreference: 'secondary' });
+
+      if (!config.usingUnifiedTopology()) {
+        this.skip();
+        return;
+      }
+
+      client.connect((err, client) => {
+        expect(err).to.not.exist;
+
+        let selectedServer;
+        const selectServerStub = sinon.stub(client.topology, 'selectServer').callsFake(function() {
+          const args = Array.prototype.slice.call(arguments);
+          const originalCallback = args.pop();
+          args.push((err, server) => {
+            selectedServer = server;
+            originalCallback(err, server);
+          });
+
+          return client.topology.selectServer.wrappedMethod.apply(this, args);
+        });
+
+        const collection = client.db().collection('test_read_preference');
+        collection.find().toArray((err, docs) => {
+          expect(err).to.not.exist;
+          expect(docs).to.have.length.greaterThan(0);
```

Comment:
Why should this pass?

Reply:
oops nice catch, originally I was inserting a document first, but we don't really care about what documents are in there. The test runs on Travis are proving that these are indeed run there.

Reply:
true, I missed the `!config.usingUnifiedTopology()` skip.